homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -248,18 +244,22 @@ async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowRes
             )
         )
 
-        if serial_number := discovery_info.upnp.get(ssdp.ATTR_UPNP_SERIAL):
-            await self.async_set_unique_id(serial_number)
-            self._abort_if_unique_id_configured()
-        else:
-            await self._async_handle_discovery_without_unique_id()
-
-        user_input = {CONF_URL: url}
+        unique_id = discovery_info.upnp.get(
+            ssdp.ATTR_UPNP_SERIAL, discovery_info.upnp[ssdp.ATTR_UPNP_UDN]
+        )
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured(updates={CONF_URL: url})
 
-        self.context["title_placeholders"] = {
-            CONF_NAME: discovery_info.upnp.get(ssdp.ATTR_UPNP_FRIENDLY_NAME)
-        }
-        return await self._async_show_user_form(user_input)
+        self.context.update(
+            {
+                "title_placeholders": {
+                    CONF_NAME: discovery_info.upnp.get(ssdp.ATTR_UPNP_FRIENDLY_NAME)
+                },
+                CONF_MANUFACTURER: discovery_info.upnp.get(ssdp.ATTR_UPNP_MANUFACTURER),
+                CONF_URL: url,
+            }
+        )
+        return await self._async_show_user_form()
```

Comment:
Side note: We should not move directly to another step's form. The form is private to the step method. Await the step method to move to that step.